### PR TITLE
[SPARK-30278][SQL][DOC] Update Spark SQL document menu for new changes

### PR DESCRIPTION
**Pull request by xuanyuanking**
### What changes were proposed in this pull request?
Update the Spark SQL document menu and join strategy hints.

### Why are the changes needed?
- Several new changes in the Spark SQL document didn't change the menu-sql.yaml correspondingly.
- Update the demo code for join strategy hints. 

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Document change only.

### CODE REVIEW

**Comment by xuanyuanking** [comment]
cc @cloud-fan @gatorsmile

File: docs/sql-performance-tuning.md [hunk 1]
```diff
@@ -158,7 +155,7 @@ broadcast(spark.table("src")).join(spark.table("records"), "key").show()
 {% highlight r %}
 src <- sql("SELECT * FROM src")
 records <- sql("SELECT * FROM records")
-head(join(broadcast(src), records, src$key == records$key))
```
Comment (cloud-fan) [inline docs/sql-performance-tuning.md]: did we remove the `broadcast` method?

**Comment by maropu** [comment]
How about adding the docs about the SQL case, too? https://github.com/apache/spark/pull/25464

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (xuanyuanking) [reply docs/sql-performance-tuning.md]: No, the `broadcast` method still available:
https://github.com/apache/spark/blob/18431c7baaba72539603814ef1757650000943d5/R/pkg/R/DataFrame.R#L4129-L4134

**Comment by xuanyuanking** [comment]
@maropu We have the SQL hint here: https://github.com/apache/spark/pull/26917/files#diff-ecc0a33250015b4a0fb4a5f1de5cc502R167, do you mean add demo for partition by hint?

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (cloud-fan) [reply docs/sql-performance-tuning.md]: so what's the recommended API to add hint? `df.hint(...)` or `hint(df, ...)`?

Comment (xuanyuanking) [reply docs/sql-performance-tuning.md]: hint(df, ...) is for R, there's no df.hint(...) in R？

**Comment by maropu** [comment]
> do you mean add demo for partition by hint?

Yea, yes. we can accept `REPARTITION_BY_RANGE` and `REPARTITION` now.

File: docs/sql-performance-tuning.md (hunk 1 above)
Comment (cloud-fan) [reply docs/sql-performance-tuning.md]: ah nvm, didn't realize it's R code

**Comment by xuanyuanking** [comment]
Sure, done in accf45c, please take a look.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by xuanyuanking** [comment]
Thanks!

**Comment by maropu** [comment]
Thanks for the update, @xuanyuanking !

**Pull request closed without merging**